### PR TITLE
OpenCV: Enable 3.3 for iOS. Drop ENABLE_NEON for iOS. Detected by __A…

### PR DESCRIPTION
**Pull request by xsacha**
…RM_NEON already.

### CODE REVIEW

**Comment by ruslo** [comment]
Testing OpenCV:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2260
* https://travis-ci.org/ingenue/hunter/builds/268901389?utm_source=github_status&utm_medium=notification

Testing OpenCV-extra:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2261
* https://travis-ci.org/ingenue/hunter/builds/268901509?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
`ios-nocodesign-9-3-arm64` toolchain failed on my local machine with error:
```
Undefined symbols for architecture arm64:
  "_WebPEncodeLosslessBGRA", referenced from:
      cv::WebPEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in libopencv_imgcodecs.a(grfmt_webp.o)
  "_WebPEncodeLosslessBGR", referenced from:
      cv::WebPEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in libopencv_imgcodecs.a(grfmt_webp.o)
  "_WebPEncodeBGRA", referenced from:
      cv::WebPEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in libopencv_imgcodecs.a(grfmt_webp.o)
  "_WebPEncodeBGR", referenced from:
      cv::WebPEncoder::write(cv::Mat const&, std::__1::vector<int, std::__1::allocator<int> > const&) in libopencv_imgcodecs.a(grfmt_webp.o)
  "_WebPGetFeaturesInternal", referenced from:
      cv::WebPDecoder::checkSignature(cv::String const&) const in libopencv_imgcodecs.a(grfmt_webp.o)
      cv::WebPDecoder::readHeader() in libopencv_imgcodecs.a(grfmt_webp.o)
  "_WebPDecodeBGRInto", referenced from:
      cv::WebPDecoder::readData(cv::Mat&) in libopencv_imgcodecs.a(grfmt_webp.o)
  "_WebPDecodeBGRAInto", referenced from:
      cv::WebPDecoder::readData(cv::Mat&) in libopencv_imgcodecs.a(grfmt_webp.o)
```

For some reason iOS build trying to use `/usr/local/lib/libwebp.dylib` from system.

**Comment by ruslo** [comment]
> Should it be using CMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY?

May be it will work but since we are using package manager the dependency that can't be handled by Hunter should not be searched.

I expect this pattern:
```cmake
option(OPENCV_WITH_WEBP "...." ON)

if(OPENCV_WITH_WEBP)
  if(HUNTER_ENABLED)
    message(FATAL_ERROR "webp is not available in Hunter")
  endif()
  find_package(webp ...)
endif()
```

**Comment by xsacha** [comment]
What if I want to include webpage library inside my project, for example, to use latest git version (not necessarily webpage but could be any package). If I use Hunter, I am then restricted to only hunter packages as dependencies?

**Comment by ruslo** [comment]
> If I use Hunter, I am then restricted to only hunter packages as dependencies?

Kind of:
* https://docs.hunter.sh/en/latest/faq/interaction-with-other-package-managers.html

As a workaround you can use `find_package` for regular system packages but as you can see this lead to inconsistency and break reproducibility (builds fine on your machine but not on mine because of `/usr/local/lib/libwebp.dylib`).

> Anyway, in this situation, OpenCV provides WebP 0.6 source code in its project. If you set BUILD_WEBP, it will build it from source instead.

I think it should be used by default in Hunter.

**Comment by xsacha** [comment]
OK.
I also noticed a problem when JPEG is enabled in OpenCV and it uses Hunter's jpeg library instead of OpenCV's:


`Build/Jpeg/toolchain.cmake -DCMAKE_MAKE_PROGRAM=/usr/bin/xcodebuild -DCMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH=NO -GXcode /Users/sacha/.hunter/_Base/414d6d8/22daf9b/301a2ff/Build/Jpeg/Source
loading initial cache file /Users/sacha/.hunter/_Base/414d6d8/22daf9b/301a2ff/cache.cmake
loading initial cache file /Users/sacha/.hunter/_Base/414d6d8/22daf9b/301a2ff/Build/Jpeg/args.cmake
-- [polly] Used toolchain: iOS 10.3 Universal (iphoneos) / clang / c++11 support
-- The C compiler identification is AppleClang 8.1.0.8020042
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -- works
-- Detecting C compile features
-- Detecting C compile features - done
-- C Compiler guessed by cmake: AppleClang
...
-- Check size of size_t
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.9/Modules/CheckTypeSize.cmake:165 (message):
  CHECK_TYPE_SIZE found different results, consider setting
  CMAKE_OSX_ARCHITECTURES or CMAKE_TRY_COMPILE_OSX_ARCHITECTURES to one or no
  architecture !
Call Stack (most recent call first):
  /Applications/CMake.app/Contents/share/cmake-3.9/Modules/CheckTypeSize.cmake:239 (__check_type_size_impl)
  CMakeLists.txt:68 (check_type_size)
`

**Comment by ruslo** [comment]
> I also noticed a problem when JPEG is enabled in OpenCV

I guess JPEG should be fixed.

**Comment by xsacha** [comment]
I fixed JPEG by just removing the size checks. It doesn't look like they are actually used?

**Comment by ruslo** [comment]
For reference: https://github.com/hunter-packages/jpeg/pull/1

File: cmake/projects/OpenCV/hunter.cmake [hunk 1]
```diff
@@ -17,6 +17,17 @@ hunter_add_version(
     PACKAGE_NAME
     OpenCV
     VERSION
+    "3.3.0-p1"
+    URL
+    "https://github.com/hunter-packages/opencv/archive/v3.3.0-p1.tar.gz"
+    SHA1
+  abab7f4244acfb2b462b788b732eafbb51d9741
```
Comment (ruslo) [inline cmake/projects/OpenCV/hunter.cmake]: Some special symbol stuck here:
```
    "https://github.com/hunter-packages/opencv/archive/v3.3.0-p1.tar.gz"
    SHA1
  abab7f4244acfb2b462b788b732eafbb51d9741
)
```

**Comment by xsacha** [comment]
No idea how that happened, but I'll just move on to P2.

**Comment by ruslo** [comment]
Testing.

OpenCV:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2318
* https://travis-ci.org/ingenue/hunter/builds/271128875?utm_source=github_status&utm_medium=notification

OpenCV-extra:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2319
* https://travis-ci.org/ingenue/hunter/builds/271129056?utm_source=github_status&utm_medium=notification

**Comment by xsacha** [comment]
Uh oh, the JPEG type size check removal seems to adversely affected the headers?

**Comment by ruslo** [comment]
> Uh oh, the JPEG type size check removal seems to adversely affected the headers?

Have no idea, probably.

**Comment by xsacha** [comment]
Reverted Jpeg since that change seems to be breaking everything.
I'm now worried that CheckTypeSize does more than I thought. Any ideas about what it is actually doing and how that could possibly break other packages?

**Comment by ruslo** [comment]
> Reverted Jpeg since that change seems to be breaking everything

Revert Jpeg commit first (Jpeg repository), then separate pull request with Jpeg version (Hunter), then this pull request.

> Any ideas about what it is actually doing and how that could possibly break other packages?

https://github.com/hunter-packages/jpeg/pull/1#pullrequestreview-60383500

**Comment by xsacha** [comment]
Should be all good to go now that jpeg has been changed.

**Comment by ruslo** [comment]
Waiting for JPEG upload:
* https://travis-ci.org/ingenue/hunter/builds/272172563?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2350
* https://travis-ci.org/ingenue/hunter/builds/272381382?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
> Testing

https://github.com/ruslo/hunter/issues/1021

**Comment by ruslo** [comment]
Testing OpenCV:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2354
* https://travis-ci.org/ingenue/hunter/builds/274518999?utm_source=github_status&utm_medium=notification

Testing OpenCV-extra:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2355
* https://travis-ci.org/ingenue/hunter/builds/274519206?utm_source=github_status&utm_medium=notification

File: cmake/projects/OpenCV/hunter.cmake [hunk 2]
```diff
@@ -17,6 +17,17 @@ hunter_add_version(
     PACKAGE_NAME
     OpenCV
     VERSION
+    "3.3.0-p2"
+    URL
+    "https://github.com/hunter-packages/opencv/archive/v3.3.0-p2.tar.gz"
+    SHA1
+    3af986b1751dd228c0eff12b16ef68af5949a4c0
```
Comment (ruslo) [inline cmake/projects/OpenCV/hunter.cmake]: Should be `27be59f7978a4625401a1041af8d108a47ae745b`

**Comment by xsacha** [comment]
That was unexpected. Not sure how that happened. Oh well :).

**Comment by ruslo** [comment]
> That was unexpected. Not sure how that happened. Oh well :)

Just for your info: https://github.com/ruslo/hunter/issues/1032#issuecomment-329116284

**Comment by xsacha** [comment]
Does it work now or we holding off until we know why the hashes are changing?

**Comment by ruslo** [comment]
Testing OpenCV:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2362
* https://travis-ci.org/ingenue/hunter/builds/275001554?utm_source=github_status&utm_medium=notification

Testing OpenCV-extra:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2363
* https://travis-ci.org/ingenue/hunter/builds/275001775?utm_source=github_status&utm_medium=notification

**Comment by xsacha** [comment]
Looks like tests passed except for when they died for other reasons (eg. log length exceeded).

Is this OK for merge?

**Comment by ruslo** [comment]
I'm still testing:
* https://travis-ci.org/ingenue/hunter/builds/276461937

Will let you know when finished.

**Comment by ruslo** [comment]
Uploading OpenCV:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2407
* https://travis-ci.org/ingenue/hunter/builds/276521790?utm_source=github_status&utm_medium=notification

Uploading OpenCV-extra:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2408
* https://travis-ci.org/ingenue/hunter/builds/276522899?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.104

**Pull request merged**